internal/providers/msentraid/msentraid.go

```diff
@@ -186,35 +186,75 @@ func (p Provider) getGroups(token *oauth2.Token, msgraphHost string) ([]info.Gro
 	}
 
 	var groups []info.Group
+	var msGroupNames []string
```

Comment:
Wouldn't be better to use a set for this `e.g. a map[string]struct{}`? And maybe use `unicode.SimpleFold` to add both cases there?

Reply:
To be able to include the unfolded group names in the log message, we need to store it unfolded. We could use a `map[string]string` for that, but tbh I'm not sure the performance benefit is worth the effort

Reply:
> Wouldn't be better to use a set for this e.g. a map[string]struct{}?

Or do you want to store the unfolded name in that map? Then we would still need to iterate over it so there is no performance benefit

Reply:
Ok, fair enough.